### DISCUSSION TITLE
Does Skosmos require `skos:hasTopConcept` relations?

### DISCUSSION THREAD

**Discussion by namedgraph** [NatLibFi/Skosmos#1573] [Q&A] [answered]
Hi,

I'm trying to reuse our taxonomy data as Linked Data and `skos:hasTopConcept` properties are problematic because the concept scheme descriptions become huge.

Does Skosmos require these properties? Looking at the query building code [here](https://github.com/NatLibFi/Skosmos/blob/master/model/sparql/GenericSparql.php#L1922) it seems that the inverse `skos:topConceptOf` on individual concepts should be enough?

**Comment by namedgraph** [comment]
@osma just need a confirmation here ;)

**Comment by osma** [comment, marked as answer]
According to the documentation, both hasTopConcept and topConceptOf are required; see https://github.com/NatLibFi/Skosmos/wiki/Data-Model#concept-schemes

But in practice, no, it's not required. There is [only one SPARQL query](https://github.com/NatLibFi/Skosmos/blob/master/model/sparql/GenericSparql.php#L624) that refers to hasTopConcept, and only for excluding it.

**Reply by namedgraph** [reply]
Thanks!
